### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Somdotta07**
### Project requirements
- Created forms to perform the following functions:
    - Created a Post on behalf of the current_user.
    - Created a comment on behalf of the current_user.

- Allowed Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    else
+      render :new, alert: 'Error occured!'
+    end
+  end
+
+  # private
+
+  # def comment_params
+  #   params.require(:comment).permit(:text)[:text]
+  # end
```
Comment (Dipeshtwis) [inline app/controllers/comments_controller.rb]: - [x] I would recommend including the private method, it makes the application secure by permitting only limited attributes.

- [x] Also add `new` method for CommentsController like PostsController. Also add `new` path for comments in `routes.rb`

```ruby
class CommentsController < ApplicationController
 def new
    @comment = Comment.new
  end
  ....
end
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -9,4 +9,30 @@ def show
...
+        end
+      end
+    end
+  end
+
+  # private
+
+  # def post_params
+  #   params.require(:data).permit(:title, :text)
+  # end
```
Comment (Dipeshtwis) [inline app/controllers/posts_controller.rb]: - [x] I would recommend including the private method, it makes the application secure by permitting only limited attributes.

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+<div class="user_con">
 <div class="user_data_post"> 
-  <h2>#<%= @post.title %> by <span><%= @post.user.name %></span></h2>
-  <h3 class = "post-comment">Comment <%= @post.comments_counter%>,  <span>Likes <%= @post.likes_counter%></span></h3>
-  <p><%= @post.text %></p>  
+  <h2><%= @post.title %> by <span><%= @post.user.name %></span></h2>
+  <%= form_with url: "/posts/#{@post.id}/likes", scope: :like do |form| %>
+    <%= form.submit "Like ", class: "post-like-btns"%>
+  <% end %>
+  <h3>Comment <%= @post.comments_counter%>,  <span>Likes <%= @post.likes_counter%></span></h3>
```
Comment (Dipeshtwis) [inline app/views/posts/show.html.erb:7]: - [x] Currently, I have added only two comments as you can see below screenshot but it's showing 4. Please fix that.

Screenshot from 2022-02-25 02-00-01

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,30 +1,47 @@
...
     <div class="user_data">
-        <div class="user_img">
-      <%= image_tag("Tom.png", width:'150px', height:'170px') %>
-    </div>
+
       <h2><%= @user.name%></h2>
+
       <p>Number of posts: <%= @user.posts_counter %></p>
+
     </div>
```
Comment (Dipeshtwis) [inline app/views/posts/index.html.erb:9]: - [x] Also with the number of posts. I have added only two posts as you can see below screenshot but it's showing 4. Please fix that.

Screenshot from 2022-02-25 02-03-24

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Somdotta07, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 3 above)
Comment (Somdotta07) [reply app/views/posts/show.html.erb:7]: Thank you for pointing out that error. Fixed this.

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Somdotta07) [reply app/controllers/posts_controller.rb]: Thank you for pointing out that error. Fixed this.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Somdotta07) [reply app/controllers/comments_controller.rb]: Thank you for your suggestion. Fixed those errors.

**Comment by Somdotta07** [comment]
Fixed errors suggested by the previous reviewer.

**Review by DammyShittu — APPROVED** [review]
Hi @Somdotta07, 👋  

# APPROVED 🟢 

_Outstanding implementation of the features in this milestone and changes suggested by the previous reviewer.✔️_

Kudos to you 💯 

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

CONGRATULATIONS! 🎉 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

# Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**